### ISSUE TITLE
linuxPackages.perf fails to build on unstable

### ISSUE THREAD

**Issue by anderspapitto** [NixOS/nixpkgs#27270]
```
$ nixos-version
17.09.git.0011f90 (Hummingbird)
```

possibly due to changes to default compiler warnings or something, because it seems to be a -Werror problem

```
util/probe-event.c: In function 'post_process_module_probe_trace_events':
util/probe-event.c:669:9: error: implicit declaration of function 'post_process_probe_trace_point' [-Werror=implicit-function-declaration]
   ret = post_process_probe_trace_point(&tevs[i].point,
         ^
util/probe-event.c:669:3: error: nested extern declaration of 'post_process_probe_trace_point' [-Werror=nested-externs]
   ret = post_process_probe_trace_point(&tevs[i].point,
   ^
cc1: all warnings being treated as errors
mv: cannot stat 'util/.probe-event.o.tmp': No such file or directory
```

**Comment by vcunat** [comment]
This seems to be an upstream problem in linux-4.9.36: https://lkml.org/lkml/2017/7/5/537

**Comment by vcunat** [comment]
Fixed by 954c66983df3.
